tunnel.go

```diff
@@ -149,10 +92,12 @@ func getAvailablePort() (int, error) {
 	if err != nil {
 		return 0, microerror.Mask(err)
 	}
-	return port, microerror.Mask(err)
+
+	return port, nil
 }
 
 // setConfigDefaults is copied and adjusted from client-go core/v1.
+// TODO what is different here? Why do we need it at all?
 func setConfigDefaults(config *rest.Config) error {
```

Comment:
Anybody?

Reply:
@fgimenez did you add this? Can you say something here? I asked a question.

Reply:
> @fgimenez did you add this? Can you say something here? I asked a question.

Nope, according to `git blame` it was added by @kopiczko, i can't help here.

Reply:
Ack. Thank you.

Reply:
We need to set schema/groupversion/deserializer magic, otherwise we can't use raw rest.Config to create a client. It will simply fail. The client is used to call `pods` resource which is in `core` api which is denoted by empty string and `v1` due to legacy reasons. setting it to other `GroupVersion` would fail at runtime.

Reply:
Why don't we configure the client as we do it in `operatorkit`? Or the other way around, why don't we do this hack we have here in `operatorkit` so we do not need to configure a specific client. Or do I get this wrong?

Reply:
`operatorkit` is more tricky. It uses different object from different groups. I think long term `operatorkit` should do this job one way or another possibly using discovery API.